qa-src/Controllers/User/UserPosts.php

```diff
@@ -27,243 +27,329 @@
-			throw new PageNotFoundException();
-		}
+		$loggedInUserId = qa_get_logged_in_userid();
 
+		list($userPoints, $questions) = $this->fetchAnswersDataFromDb($handle, $loggedInUserId, $profileUserId);
 
 		// Get information on user questions
 
 		$pagesize = qa_opt('page_size_activity');
-		$count = (int)@$userpoints['aposts'];
+		$count = (int)@$userPoints['aposts'];
 		$questions = array_slice($questions, 0, $pagesize);
-		$usershtml = qa_userids_handles_html($questions, false);
-
+		$usersHtml = qa_userids_handles_html($questions, false);
 
 		// Prepare content for theme
 
 		$qa_content = qa_content_prepare(true);
 
-		if (count($questions))
-			$qa_content['title'] = qa_lang_html_sub('profile/answers_by_x', $this->userhtml);
-		else
-			$qa_content['title'] = qa_lang_html_sub('profile/no_answers_by_x', $this->userhtml);
-
+		$qa_content['title'] = qa_lang_html_sub(
+			empty($questions) ? 'profile/no_answers_by_x' : 'profile/answers_by_x',
+			$this->userHtml($handle, $profileUserId)
+		);
 
 		// Recent questions by this user
 
-		$qa_content['q_list']['form'] = array(
-			'tags' => 'method="post" action="' . qa_self_html() . '"',
-
-			'hidden' => array(
-				'code' => qa_get_form_security_code('vote'),
-			),
-		);
+		$this->addForm($qa_content);
 
 		$qa_content['q_list']['qs'] = array();
 
-		$htmldefaults = qa_post_html_defaults('Q');
-		$htmldefaults['whoview'] = false;
-		$htmldefaults['avatarsize'] = 0;
-		$htmldefaults['ovoteview'] = true;
-		$htmldefaults['answersview'] = false;
+		$htmlDefaults = qa_post_html_defaults('Q');
+		$htmlDefaults['whoview'] = false;
+		$htmlDefaults['avatarsize'] = 0;
+		$htmlDefaults['ovoteview'] = true;
+		$htmlDefaults['answersview'] = false;
+
+		$voteView = qa_get_vote_view('A', false, false);
 
 		foreach ($questions as $question) {
-			$options = qa_post_html_options($question, $htmldefaults);
-			$options['voteview'] = qa_get_vote_view('A', false, false);
+			$options = qa_post_html_options($question, $htmlDefaults);
+			$options['voteview'] = $voteView;
+
+			$qa_content['q_list']['qs'][] = qa_other_to_q_html_fields(
+				$question,
+				$loggedInUserId,
+				qa_cookie_get(),
+				$usersHtml,
+				null,
+				$options
+			);
+		}
+
+		$this->addPageLinks($qa_content, $pagesize, $count);
+		$this->addNavigation($qa_content, $handle, $loggedInUserId, $profileUserId, 'answers');
+
+		return $qa_content;
+	}
+
+	/**
+	 * Return the HTML to display for the handle, and if we're using external users, determine the userid.
+	 *
+	 * @param string $handle
+	 * @param mixed $userId
+	 *
+	 * @return mixed|string
+	 */
+	private function userHtml($handle, $userId)
+	{
+		if (QA_FINAL_EXTERNAL_USERS) {
+			$usersHtml = qa_get_users_html(array($userId), false, qa_path_to_root(), true);
 
-			$qa_content['q_list']['qs'][] = qa_other_to_q_html_fields($question, $loginuserid, qa_cookie_get(),
-				$usershtml, null, $options);
+			return @$usersHtml[$userId];
```

Comment:
You should use `try..catch` statement

Reply:
Exactly what exception are you suggesting to catch in that line?

Reply:
Hmm... 
I did not look exactly at this code, but in the code we should not use `@` - if I see such a sign next to a variable / function, I always try to rewrite the code so that the character does not exist.